File: source/includes/write/insert.php
<?php

require 'vendor/autoload.php';

$uri = getenv('MONGODB_URI') ?: throw new RuntimeException('Set the MONGODB_URI variable to your Atlas URI that connects to the sample dataset');
$client = new MongoDB\Client($uri);

// start-db-coll
$collection = $client->sample_restaurants->restaurants;
// end-db-coll

// Inserts a document that stores a "name" value of "Mongo's Burgers" into the collection
// start-insert-one
$result = $collection->insertOne(['name' => 'Mongo\'s Burgers']);
// end-insert-one

// Inserts documents representing restaurants into the collection
// start-insert-many
$restaurants = [
    ['name' => 'Mongo\'s Burgers'],
    ['name' => 'Mongo\'s Pizza']
];

$result = $collection->insertMany($restaurants);
// end-insert-many

// Inserts multiple documents and instructs the insert operation to skip document-level validation
// start-modify
$docs = [
    ['name' => 'Mongo\'s Burgers'],
    ['name' => 'Mongo\'s Pizza'],
    ['name' => 'Mongo\'s Tacos']
];

$result = $collection->insertMany($docs, ['bypassDocumentValidation' => true]);
// end-modify

File: source/includes/write/insert.scala

import org.mongodb.scala._
import org.mongodb.scala.bson.Document
import org.mongodb.scala.model._
import org.mongodb.scala.result._

object Insert {

  def main(args: Array[String]): Unit = {
    val mongoClient = MongoClient("<connection string URI>")

    // start-db-coll
    val database: MongoDatabase = mongoClient.getDatabase("sample_restaurants")
    val collection: MongoCollection[Document] = database.getCollection("restaurants")
    // end-db-coll

    // start-insert-one
    val doc: Document = Document("name" -> "Neighborhood Bar & Grill", "borough" -> "Queens")
    val observable: Observable[InsertOneResult] = collection.insertOne(doc)

    observable.subscribe(new Observer[InsertOneResult] {
      override def onNext(result: InsertOneResult): Unit = println(result)
      override def onError(e: Throwable): Unit = println("Failed: " + e.getMessage)
      override def onComplete(): Unit = println("Completed")
    })
    // end-insert-one

    // start-insert-many
    val docs: Seq[Document] = Seq(
      Document("name" -> "Metropolitan Cafe", "borough" -> "Queens"),
      Document("name" -> "Yankee Bistro", "borough" -> "Bronx")
    )
    val observable: Observable[InsertManyResult] = collection.insertMany(docs)

    observable.subscribe(new Observer[InsertManyResult] {
      override def onNext(result: InsertManyResult): Unit = println(result)
      override def onError(e: Throwable): Unit = println("Failed: " + e.getMessage)
      override def onComplete(): Unit = println("Completed")
    })
    // end-insert-many

    // start-insert-opts
    val docs: Seq[Document] = Seq(
      Document("name" -> "One Night's Delight", "borough" -> "Queens"),
      Document("name" -> "Second Street Pub", "borough" -> "Manhattan"),
      Document("name" -> "Triple Crown Diner", "borough" -> "Brooklyn")
    )
    val opts: InsertManyOptions = InsertManyOptions().bypassDocumentValidation(true)
    val observable: Observable[InsertManyResult] = collection.insertMany(docs, opts)

    observable.subscribe(new Observer[InsertManyResult] {
      override def onNext(result: InsertManyResult): Unit = println(result)
      override def onError(e: Throwable): Unit = println("Failed: " + e.getMessage)
      override def onComplete(): Unit = println("Completed")
    })
    // end-insert-opts

    Thread.sleep(1000)
    mongoClient.close()
  }
}
